cmd/stacker/internal_go.go

```diff
@@ -208,3 +218,51 @@ func doAtomfsUmount(ctx *cli.Context) error {
 	mountpoint := ctx.Args().Get(0)
 	return atomfs.Umount(mountpoint)
 }
+
+// pkgname, license, paths...
+func doBomBuild(ctx *cli.Context) error {
+	if ctx.Args().Len() < 7 {
+		return errors.Errorf("wrong number of args")
+	}
+
+	dest := ctx.Args().Get(0)
+	author := ctx.Args().Get(1)
+	org := ctx.Args().Get(2)
+	license := ctx.Args().Get(3)
+	pkgname := ctx.Args().Get(4)
+	pkgversion := ctx.Args().Get(5)
+	paths := []string{}
+	for i := 6; i < ctx.Args().Len(); i++ {
+		paths = append(paths, ctx.Args().Get(i))
+	}
+	out := path.Join(dest, fmt.Sprintf("doc-%s.spdx.json", pkgname))
+	name := fmt.Sprintf("doc-%s", pkgname)
+
+	return fs.BuildPackage(name, author, org, license, pkgname, pkgversion, paths, out)
+}
+
+func doBomVerify(ctx *cli.Context) error {
+	if ctx.Args().Len() != 4 {
+		return errors.Errorf("wrong number of args")
+	}
+
+	// merge
+	dest := ctx.Args().Get(0)
+	name := ctx.Args().Get(1)
+	author := ctx.Args().Get(2)
+	org := ctx.Args().Get(3)
+
+	if err := bom.MergeDocuments("/stacker-artifacts", name, author, org, dest); err != nil {
```

Comment:
it is weird that we merge all of the things under the generically named '/stacker-artifacts' (artifacts) as if they are all 'bom').  Do we need to namespace that further, so that the directory would be /stacker-artifacts-bom ? As i mention below, i think ideally a single top level directory 'stacker' is prefereable, so /stacker/artifacts/bom ... all the files in that directory would be expected to be 'bom'.

Reply:
Meant to catch all artifacts - bom is one, we have an inventory file also. No need to complicate the convention unnecessarily?

Reply:
No need to complicate, but this is the kind of thing that once established can be painful (both to us and users) to extend later, if you decide there are other artifacts you want to deal with.  /stacker-artifacts/bom looks good to me.  (As evidence of the possible dangers, /stacker/artifacts unnecessarily risks conflicting with a user's imports)

Reply:
You're right, that /stacker/artifacts does risk collision with an import from a user 'stacker.yaml' in a pre-1.0 release of stacker.  I think thats probably very low likelyhood.

The problem I have with /stacker-artifacts is that, the next time we have a feature that needs a path inside the container, you'll follow the convention and use /stacker-other-thing.

That would mean we'd have:

    /stacker
    /stacker-artifatcs
    /stacker-other-thing
    /stacker-thing-a

So if a user is to stay out of stacker's potential namespace, they have to just avoid any paths that start with /stacker.  The above just unplanned to anyone thinking about it.

I think the right thing to do is to tell users the whole tree is theirs except /stacker. 

To do that cleanly, it would make sense to then move the imports into /stacker/imports. 

It really feels like just biting the bullet and doing this is the right way forward.

Reply:
We currently only have 1) /stacker (imported), 2) /stacker-artifacts (layer related artifacts), 3) /stacker-bom (embedded binary)
In "future" expect things like in-toto attestations etc which will also land in 2)

Reply:
Are you  willing to guarantee to the user that /stacker/ and /stacker-artifacts/ are the only paths that stacker will interact with?  As a user, i'd like to know that whatever (silly or problematic paths) my container has, stacker will not break or pollute my container.

As a developer, I think it is much more future-safe to adjust the top level /stacker/ now and promise that stacker will not touch any other paths. It is OK for stacker to touch paths that are explicitly requested by the user as through 'import' with a dest or other explicit mechanism.

Reply:
`import:` is a directive that can possibly collide

So here is the scheme:
/stacker/stacker-bom <- `/proc/self` bind-mounted into container
/stacker/imports/<imported> <- read-only non-dest imports (this is a breaking change)
/stacker/artifacts <- writable from inside the container

Reply:
i would suggest moving 'stacker-bom' into /stacker/tools or /stacker/bin

Reply:
(revision) /stacker/tools/stacker-bom